src/snowflake/cli/api/config.py

```diff
@@ -126,16 +175,23 @@ def config_section_exists(*path) -> bool:
         return False
 
 
-def get_connection(connection_name: str) -> dict:
+def get_all_connections() -> dict[str, ConnectionConfig]:
+    return {
+        k: ConnectionConfig.from_dict(connection_dict)
+        for k, connection_dict in get_config_section("connections").items()
+    }
+
+
+def get_connection_dict(connection_name: str) -> dict:
```

Comment:
I don't want to do too many changes in this PR (to keep it relatively safe), so I decided to keep other usages of "get connection" still as dict. This is the place for further refactoring -> changing it to ConnectionConfig.